pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -0,0 +1,530 @@
+		},
+		Spec: hivev1.SyncSetSpec{
+			ClusterDeploymentRefs: []corev1.LocalObjectReference{
+				{
+					Name: clusterDeploymentName,
+				},
+			},
+			SyncSetCommonSpec: hivev1.SyncSetCommonSpec{
+				ResourceApplyMode: hivev1.SyncResourceApplyMode,
+				Secrets: []hivev1.SecretMapping{
+					{
+						SourceRef: hivev1.SecretReference{
+							Name:      dmsSecret,
+							Namespace: namespace,
+						},
+						TargetRef: hivev1.SecretReference{
+							Name:      dmsi.Spec.TargetSecretRef.Name,
+							Namespace: dmsi.Spec.TargetSecretRef.Namespace,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return newSS
+
+}
+
+// delete snitches,secrets and syncset associated with the dmsi that has been deleted
+func (r *ReconcileDeadmansSnitchIntegration) deleteDMSI(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, clustDeploy *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", clustDeploy.Name, "cluster-deployment.Namespace:", clustDeploy.Namespace)
+	var DeadMansSnitchFinalizer string = "dms.managed.openshift.io/deadmanssnitch-" + dmsi.Name
+
+	// Delete the dms
+	logger.Info("Deleting the DMS from api.deadmanssnitch.com")
+	snitchName := clustDeploy.Spec.ClusterName + "." + clustDeploy.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
+	snitches, err := dmsc.FindSnitchesByName(snitchName)
+	if err != nil {
+		return err
+	}
+	for _, s := range snitches {
+		delStatus, err := dmsc.Delete(s.Token)
+		if !delStatus || err != nil {
+			logger.Info("Failed to delete the DMS from api.deadmanssnitch.com")
+			return err
+		}
+		logger.Info("Deleted the DMS from api.deadmanssnitch.com")
+	}
+
+	// Delete the SyncSet
+	logger.Info("Deleting DMS SyncSet")
+	err = utils.DeleteSyncSet(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting SyncSet")
+		return err
+	}
+
+	// Delete the referenced secret
+	logger.Info("Deleting DMS referenced secret")
+	err = utils.DeleteRefSecret(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting secret")
+		return err
+	}
+
+	logger.Info("Deleting DMS finalizer from dmsi")
+	if utils.HasFinalizer(dmsi, DeadMansSnitchFinalizer) {
+		utils.DeleteFinalizer(dmsi, DeadMansSnitchFinalizer)
+		err = r.client.Update(context.TODO(), dmsi)
+		if err != nil {
+			logger.Error(err, "Error deleting Finalizer from dmsi")
+			return err
+		}
+
+		return nil
+	}
+	return nil
+
+}
+
+// delete snitches,secrets and syncset associated with the cluster deployment that has been deleted
+func (r *ReconcileDeadmansSnitchIntegration) deleteDMSClusterDeploy(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, clustDeploy *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
```

Comment:
The CRD name this references is `ClusterDeployment`.  I'd prefer we use the full name for the function name and function argument and not shortened.  I won't hold the review for that though.  Also acceptable is the CRD's short name "cd" but that's pretty terse.